src/agent_sdk/inc/du_agent_sdk/agent_module_interface.h

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file agent_module_interface.h
+ * @brief Defines types and methods for DU Agent module library.
+ *
+ * @copyright Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ */
+
+#ifndef ADUC_AGENT_MODULE_INTERFACE_H
+#define ADUC_AGENT_MODULE_INTERFACE_H
+
+#include "du_agent_sdk/agent_common.h"
+#include "aduc/result.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+typedef void* ADUC_AGENT_MODULE_HANDLE;
+
+typedef ADUC_AGENT_MODULE_HANDLE (*ADUC_AGENT_MODULE_CREATE)(void);
+typedef void (*ADUC_AGENT_MODULE_DESTROY)(ADUC_AGENT_MODULE_HANDLE);
+
+typedef struct ADUC_AGENT_CONTRACT_INFO_TAG
+{
+    const char* provider;
+    const char* name;
+    const int version;
+    const char* contractInfo;
+} ADUC_AGENT_CONTRACT_INFO;
+
+/**
+ * @brief The module data types
```

Comment:
Not very descriptive. Could it include why these are needed and example of modules that use which module data types with brief examples?